### PR TITLE
Fixes #1723: Problem importing graphml which was exported graphml from apoc

### LINKED ISSUE #1451: Possible bug in Neo4J 4.0 version, during GRAPHML import
## Expected Behavior (Mandatory)
Able to successfully import the graphML file to a graph DB in Neo4j

## Actual Behavior (Mandatory)
Errors out giving below exception, works in 3.x series.
```Failed to invoke procedure `apoc.import.graphml`: Caused by: org.neo4j.kernel.impl.util.collection.MemoryAllocationLimitException: Can't allocate 524288 bytes due to exceeding memory limit; used=2147028992, max=2147483648```

## How to Reproduce the Problem

```
CALL apoc.import.graphml("my_graph.graphml", {})`

```

### Steps (Mandatory)

1.  Install Neo4J Desktop
2. Modify config settings to allow apoc import
3. create a new graph DB
4. RUN below code to import the graphml
5. `CALL apoc.import.graphml("my_graph.graphml", {})`

  

## Specifications (Mandatory)

Currently used versions

### Versions

  - OS:MacOS
  - Neo4j:4.0.1
  - Neo4j-Apoc:4.0.0.4

### LINKED ISSUE #1723: Problem importing graphml which was exported graphml from apoc
## Expected Behavior (Mandatory)

Able to import large graphml files into Neo4j.

## Actual Behavior (Mandatory)
It throws this error:
Failed to invoke procedure `apoc.import.graphml`: Caused by: org.neo4j.kernel.impl.util.collection.MemoryAllocationLimitException: Can't allocate 524288 bytes due to exceeding memory limit; used=2147028992, max=2147483648

However, a student of mine did the same cypher queries in Neo4j v3 and it worked.

## How to Reproduce the Problem
Try to import a 2.9 GB graphml file

### Simple Dataset (where it's possible)

I used the neo4j database from Reactome (https://reactome.org/download-data)

### Steps (Mandatory)

  1. export with: CALL apoc.export.graphml.all("pathwaydata.graphml", {batchSize:10000, readLabels: true, storeNodeIds: false, useTypes: true})
  2. import with: CALL apoc.import.graphml("pathwaydata.graphml", {batchSize: 10000, readLabels: true, storeNodeIds:false, useTypes: true})

## Specifications (Mandatory)

Currently used versions

### Versions

  - Linux community
  - Neo4j: 4.0.3
  - Neo4j-Apoc 4.0.0.16

### PR DESCRIPTION
**Pull request by vga91**
Fixes #1723
Fixes #1451

Changed node assignment from `this.tx` to `tx.getTransaction()`.

### CODE REVIEW

**Review by fbiville — APPROVED** [review]

**Comment by fbiville** [comment]
@vga91 were you able to determine whether it'd be feasible to configure a small heap for the Neo4j test instance? That way, you'd be able to reproduce an OOM with a not-too-big data set.

**Comment by vga91** [comment]
> @vga91 were you able to determine whether it'd be feasible to configure a small heap for the Neo4j test instance? That way, you'd be able to reproduce an OOM with a not-too-big data set.

Fine, I didn't think about it. Added a test with a relatively small file (about 50mb).

File: core/src/test/java/apoc/export/graphml/ExportGraphMLTest.java [hunk 1]
```diff
@@ -182,6 +187,22 @@ public void testImportGraphML() throws Exception {
         TestUtil.testCall(db, "MATCH  (c:Bar {age: 12, values: [1,2,3]}) RETURN COUNT(c) AS c", null, (r) -> assertEquals(1L, r.get("c")));
     }
 
+    @Test
+    public void testImportGraphMLLargeFile() {
```
Comment (fbiville) [inline core/src/test/java/apoc/export/graphml/ExportGraphMLTest.java:191]: @vga91 does that test fail without your fix?

Comment (vga91) [reply core/src/test/java/apoc/export/graphml/ExportGraphMLTest.java:191]: @fbiville Exactly. Throws an `org.neo4j.kernel.impl.util.collection.MemoryAllocationLimitException` like the related issues.

**Pull request merged**